### DISCUSSION TITLE
Profiling with webassembly target

### DISCUSSION THREAD

**Discussion by ArcticXWolf** [hajimehoshi/ebiten#1876] [Q&A] [answered]
Hey,

I know this might be out of scope, but is there a nice way to do profiling (best case with the golang profiling tools) when using wasm as the target platform?

My reason is that my emulator isnt running as fast as I want it to and thus I would love to take a look at the cpu profile. However currently it is designed to only run on wasm (no cli-flags, no filesystem access, etc).

I havent been able to find much on this topic, only the following blog article and this is 1. chrome-only and 2. a bit complicated: https://agniva.me/wasm/2019/07/25/wasm-cpu-profiles.html

Greetings,
ArcticXWolf

**Comment by Zyko0** [comment]
I also need that !

**Comment by hajimehoshi** [comment, marked as answer]
You can use Chrome devtools. Have you tried? https://developer.chrome.com/docs/devtools/evaluate-performance/

**Reply by ArcticXWolf** [reply]
I dont have Chrome available at the moment, only Firefox. In Firefox, I can use their profiling too, but it does not export function names, it only says wasm_function everywhere. Is chrome able to show the real names or are they missing in the output too?

**Reply by hajimehoshi** [reply]
>  Is chrome able to show the real names or are they missing in the output too?

Yes, Chrome shows every function name:

![image](https://user-images.githubusercontent.com/16950/141689211-240c32c2-0e39-476b-b3f0-490ed709db77.png)

**Reply by ArcticXWolf** [reply]
Ahhh, nevermind...
I had my linker set to remove debug symbols... firefox can do this too..

![image](https://user-images.githubusercontent.com/5812215/141693501-868b53e0-067e-4502-9bac-7455d511401b.png)

Sorry for not seeing this earlier!
